parlai/scripts/train_model.py

```diff
@@ -715,6 +716,8 @@ def train(self):
             self.save_model()
         elif opt.get('model_file'):
             # reload best validation model
+            del self.agent
```

Comment:
I think you only need this, not the empty cache.  [from the docs](https://pytorch.org/docs/stable/cuda.html#torch.cuda.empty_cache)

![image](https://user-images.githubusercontent.com/31896/75741691-60ce1a80-5cd9-11ea-9217-e893f6f22ef4.png)